inline_test.go

```diff
@@ -570,13 +570,16 @@ func TestInlineLink(t *testing.T) {
 
 		// no closing ), autolinking detects the url next
 		"[disambiguation](http://en.wikipedia.org/wiki/Disambiguation_(disambiguation) is the",
-		"<p>[disambiguation](<a href=\"http://en.wikipedia.org/wiki/Disambiguation_(disambiguation\">http://en.wikipedia.org/wiki/Disambiguation_(disambiguation</a>) is the</p>\n",
+		"<p>[disambiguation](<a href=\"http://en.wikipedia.org/wiki/Disambiguation_(disambiguation)\">http://en.wikipedia.org/wiki/Disambiguation_(disambiguation)</a> is the</p>\n",
```

Comment:
See: http://spec.commonmark.org/0.27/#link-destination

> A link destination consists of either

>  * a sequence of zero or more characters between an opening < and a closing > that contains no spaces, line breaks, or unescaped < or > characters, or
> * a nonempty sequence of characters that does not include ASCII space or control characters, and includes parentheses only if (a) they are backslash-escaped or (b) they are part of a balanced pair of unescaped parentheses that is not itself inside a balanced pair of unescaped parentheses.

The spec does not talk about unescaped and unbalenced parenthesis in link destination. So I was just trying to make it behave the way http://spec.commonmark.org/dingus/ behaves for such links. It ignores the entire link processing in that case and renders it literally after processing. So:

`foo [disambiguation](http://en.wikipedia.org/wiki/Disambiguation_(disambiguation) is the`

becomes:

`<p>foo [disambiguation](http://en.wikipedia.org/wiki/Disambiguation_(disambiguation) is the</p>`.